mindsdb/integrations/libs/base.py

```diff
@@ -110,7 +110,75 @@ def __init__(self, name: str):
 
 class PredictiveHandler(BaseHandler):
     """
+    DEPRECATED.
+
     Base class for handlers associated to predictive systems.
     """
     def __init__(self, name: str):
         super().__init__(name)
+
+
+class BaseMLEngine:
+    """
+    Base class for integration engine to connect with other Machine Learning libraries/frameworks.
+
+    This class will be instanced when interacting with the underlying framework.
+    """
+
+    def __init__(self, model_storage, engine_storage) -> None:
+        """
+        Warning: This method should not be overridden.
+
+        Initialize storage objects required by the ML engine.
+
+        - engine_storage: persists global engine-related internals.
+        - model_storage: stores artifacts for any given model.
+        """
+        self.model_storage = model_storage
+        self.engine_storage = engine_storage
+
+    def create(self, target: str, df: Optional[pd.DataFrame] = None, args: Optional[Dict] = None) -> None:
```

Comment:
I think it is better to return a `HandlerStatusResponse` here as lightwood handler does it. We need to have an abstraction for returning data.

Reply:
Yes, however that's being done by the wrapper now (`libs/ml_exec_base.py:BaseMLEngineExec`) which enables this class to be much simpler.